### PR TITLE
LF: rename conversion builtins

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
The name of some builtins will be exposed as part of the Exception
message. This PR tries to make conversion builtins naming more 
consistent and more obvious, before we could not rename those 
anymore.

This is part of #8020

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs [hunk 1]
```diff
@@ -257,17 +257,17 @@ instance Pretty BuiltinExpr where
     BEGreaterEqNumeric -> "GEQ_NUMERIC"
     BEGreaterNumeric -> "GREATER_NUMERIC"
     BENumericFromText -> "FROM_TEXT_NUMERIC"
-    BEToTextNumeric -> "TO_TEXT_NUMERIC"
+    BEToTextNumeric -> "NUMERIC_TO_TEXT"
```
Comment (cocreature) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:260]: Can we also rename the constructors?

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs [hunk 2]
```diff
@@ -257,17 +257,17 @@ instance Pretty BuiltinExpr where
     BEGreaterEqNumeric -> "GEQ_NUMERIC"
     BEGreaterNumeric -> "GREATER_NUMERIC"
     BENumericFromText -> "FROM_TEXT_NUMERIC"
```
Comment (cocreature) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:259]: Would it be consistent to call this `NUMERIC_FROM_TEXT` or maybe `TEXT_TO_NUMERIC`?

File: daml-lf/spec/daml-lf-1.rst [hunk 3]
```diff
@@ -4805,13 +4805,13 @@ exercise_by_key
 The deserialization process will reject any Daml-LF 1.8 (or earlier)
 program using the field ``exercise_by_key`` in the ``Update`` message.
 
-TO_TEXT_CONTRACT_ID
+ID_TO_TEXT_CONTRACT
```
Comment (cocreature) [inline daml-lf/spec/daml-lf-1.rst:4808]:
```suggestion
CONTRACT_ID_TO_TEXT
```

File: daml-lf/spec/daml-lf-1.rst [hunk 4]
```diff
@@ -4805,13 +4805,13 @@ exercise_by_key
...
 ...................
 
 [*Available in versions >= 1.11*]
 
 The deserialization process will reject any Daml-LF 1.8 (or earlier)
-program using the builtin function ``TO_TEXT_CONTRACT_ID``.
+program using the builtin function ``ID_TO_TEXT_CONTRACT``.
```
Comment (cocreature) [inline daml-lf/spec/daml-lf-1.rst:4814]:
```suggestion
program using the builtin function ``CONTRACT_ID_TO_TEXT``.
```

**Review by cocreature — APPROVED** [review]
Thanks!

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs (hunk 2 above)
Comment (remyhaemmerle-da) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:259]: In another PR.

Comment (remyhaemmerle-da) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:259]: :+1:

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs (hunk 1 above)
Comment (remyhaemmerle-da) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:260]: in another PR

**Review by sofiafaro-da — APPROVED** [review]
Nice, thank you!

**Pull request merged**